File: atcoder/ABC/abc201-300/abc289/e.py
from collections import defaultdict, deque
t = int(input())

def solve(n, m, c, edges):
    visited = defaultdict(lambda: False)
    visited[1] = True
    q = deque([[1, n, 0]])
    ans = -1
    while len(q) > 0:
        takahashi_current_num, aoki_current_num, cost = q.popleft()
        if current_num == n and aoki_current_num == 1:
            ans = cost
            break
        for next_num in edges[current_num]:
            if visited[next_num]:
                continue
            if colors[current_num] == colors[next_num]:
                continue
            visited[next_num] = True
            q.append([next_num, cost + 1])
    return ans

ans = []
for _ in range(t):
    n, m = map(int, input().split())
    c_l = list(map(int, input().split()))
    colors = dict()
    for i, c_i in enumerate(c_l):
        colors[i + 1] = c_i
    uv = defaultdict(lambda: [])
    for _ in range(m):
        u, v = map(int, input().split())
        uv[u].append(v)
        uv[v].append(u)
    ans.append(str(solve(n, m, colors, uv)))
print("\n".join(ans))

File: atcoder/ABC/abc201-300/abc289/src/bin/e.rs
use proconio::input;
use std::collections::VecDeque;


fn main() {
    input! {
        t: i64
    }

    for _ in 0..t {
        input! {
            n: i64,
            m: i64,
            c: [i64; n],
            uv: [[i64; 2]; m]
        }

        // 幅優先探索
        let mut q: VecDeque<(i64, i64)> = VecDeque::new();
        // [現在地, コスト]
        q.push_back((1, 0));

        while let Some((current_edge, cost)) = q.pop_front() {
            
        }
    }


    println!("{}", ans);
}
